### ISSUE TITLE
Upload restriction for WiFi only does not work.

### ISSUE THREAD

**Issue by levster69** [nextcloud/android#1703]
Hi,

Running the updated 2.0.0 Android version, with the uploads restricted to WiFi only and charging. Did not pay attention and for the last seven days the uploads were going over both WiFi and Data, using over 15GB in just 7 days!!! How can I restrict this and enforce it?

Nextcloud server version: 12.0.3

**Comment by Marcwa19197** [comment]
i have exactly the same issue.
Only a remove of the app stops the uploading process. now i reinstalled the nextcloud v2.0 app and configured auto uploading for only WiFi. Now nothing is getting uploaded. I moved my old Media folder on my cloud where the old auto upload process saved his files, now only newly added pictures are uploaded. the older nextcloud app version uploaded all files, even if they are existing on the device..

**Comment by AndyScherzinger** [comment]
cc @mario do you have any idea? Auto Upload ignoring wifi-only restriction and burning through some users' data plans has been reported through GPlay couple of times now :(

**Comment by mario** [comment]
@AndyScherzinger there's only one possible thing I know of - you have two interfaces, one wifi and one 4g. We check if you're on WIfi, get yes, and initiate upload. I don't think we ever set an interface over which upload goes.

**Comment by AndyScherzinger** [comment]
@mario so we don't continue uploading when wifi get's turned of during an upload session? I know we had this issue in the past but also thought this got resolved.

@levster69 @Marcwa19197 what devices / android versions are you running.

> the older nextcloud app version uploaded all files, even if they are existing on the device..

Afaik we never had this implementation so if that happened it wasn't on purpose, while it is a feature we plan for future increments of the auto upload (see #285)

**Comment by Marcwa19197** [comment]
Hi,
thanks for the fast reply.
i use an Galaxy S8+ on Android 7.0 (Latest from Samsung, no root, no custom rom)

I work around to upload all existing files again: I simply moved the dcim folder to an temp directory and move it back, now the files are reuploading (and only on wifi).

i cant remember exactly, maybe i updated to nextcloud v2.0 using mobile data, so maybe the check if im connected via wifi failed or was skipped. So it started to upload >all< Files again, adding "(2)" to every filename (around 15gb) so my data plan was empty over the day (no problem - my data plan starts at 25.) the only way to stop this was by reinstalling the app completely

**Comment by mario** [comment]
We do continue, fixing this is planned for next release.

**Comment by mario** [comment]
Hey,

This couldnt have happened usinng 2.0 - with 2.0 you cannot produce
duplicate uploads.

Cheers!

**Comment by levster69** [comment]
I am running Samsung S7, Android 7.0, Kernel 3.18.31

As far as duplicates, I all of a sudden, in a matter of less than a week see a lot of them, (2), (2)(2), (2)(2)(2)(2)....

Here is one: 20171017_195452 (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2) (2).mp4 

They are all of the same size uploaded...

**Comment by mario** [comment]
delete the app, and install fresh 2.0.

**Comment by j-ed** [comment]
> @mario We do continue, fixing this is planned for next release.

If you're going to apply a fix for this it might be worse to check if you can account the enhancement request #436 too, which is related to the automatic upload action. So you would be able to hit two birds with one stone 😉

**Comment by tobiasKaminsky** [comment]
@mario I remember that there is already a fix included in master, is this correct?

**Comment by mario** [comment]
@tobiasKaminsky we did make some changes regarding this, unsure if it fixes it though.

**Comment by tobiasKaminsky** [comment]
Here we check for UNMETERED:
https://github.com/nextcloud/android/blob/8a865ee6382b05ffaca9ccd2b7e90e415befc686/src/main/java/com/owncloud/android/operations/UploadFileOperation.java#L351-L351

Doc says:
>METERED: This job requires metered connectivity such as most cellular data networks.
>UNMETERED: Network must be connected and unmetered.

--> "such as most..." maybe it happens that some 4g connections are wrongly detected as unmetered?
Maybe we can use 

https://developer.android.com/training/monitoring-device-state/connectivity-monitoring.html#DetermineType

```
ConnectivityManager cm = (ConnectivityManager)context.getSystemService(Context.CONNECTIVITY_SERVICE);
NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
boolean isWiFi = activeNetwork != null && activeNetwork.getType() == ConnectivityManager.TYPE_WIFI;
```

**Comment by mario** [comment]
@tobiasKaminsky this is what we do already :P I guess if the uploads don't pause if you move networks. That might actually be fixed, but I didn't test.

**Comment by tobiasKaminsky** [comment]
Fixed via #2025
